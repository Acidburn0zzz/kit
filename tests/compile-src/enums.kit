enum MySimpleEnum {
    Apple;
    Banana;
    Strawberry;

    // public function toString(): CString {
    //     return match this {
    //         Apple => "apple";
    //         Banana => "banana";
    //         Strawberry => "strawberry";
    //     }
    // }
}

enum MyComplexEnum {
    Apple2(i: Int);
    Banana2(a: Float, b: CString);
    Strawberry2;
}

function main() {
    var a = Apple;
    var b = Banana;
    var c = Apple2(1);
    var d = Banana2(5, "hello");
    var e = Strawberry2;

    var f: MyComplexEnum = d;
    f = e;

    if (a != b) {
        printf("hello!\n");
    }

    // printf("%s\n", a.toString());
    // printf("%s\n", b.toString());
}
