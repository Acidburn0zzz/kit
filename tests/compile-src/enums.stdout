hello!
apple
banana
Apple2: 1
Banana2: 5.0, hello
Strawberry2
